### DISCUSSION TITLE
Use squash-merge when merging PRs

### DISCUSSION THREAD

**Discussion by faridyagubbayli** [waltsims/k-wave-python#85] [General]
Most of the PRs include `work in progress` commits. I think we should avoid adding those commits to the `main` branch and use `squash-merge`. This way we will have a single commit per PR in the `main` branch. It should also help if we need to revert a particular PR in the main. What do people think?

**Comment by waltsims** [comment]
I think this makes sense :)

**Reply by waltsims** [reply]
The argument against this is that it deletes history needlessly. It also makes catch up merges potentially harder to understand.
